src/services/recordService.ts

```diff
@@ -160,125 +159,66 @@ const recordService = {
       );
   },
 
-  searchWithFallbackIfNeeded(
-    ownerId: string,
-    countOnly: boolean,
-    params: Params = {}
-  ): Promise<any[]> {
-    const searchParams = { ...params };
-
-    const appDataFlag = taggingUtils.generateAppDataFlagTag();
-    if (!params?.tags?.includes(appDataFlag)) {
-      searchParams.exclude_flags = [appDataFlag];
+  /**
+   * Encrypt a tag or a tag group with the user's tag encryption key
+   *
+   * A tag group starts with an opening bracket (_(_) and ends with a closing bracket (_)_) and
+   * includes a comma separated list of tags
+   * @param tek The tag encryption key of the user
+   * @param tag The tag or tag group to encrypt
+   * @returns The encrypted tag or tag group
+   */
+  async encryptTag(tek: SymKey, tag: string) {
+    if (tag.startsWith('(') && tag.endsWith(')')) {
+      const partials = tag.slice(1, -1).split(',');
+      return `(${(await Promise.all(partials.map(partial => symEncryptString(tek, partial)))).join(
+        ','
+      )})`;
     }
```

Comment:
I am not really understand why this is necessary - I mean you could simply encrypt before you craft the tagGroups, right? Or do I miss something?